Framework/Algorithms/src/FFTSmooth2.cpp

```diff
@@ -207,34 +205,28 @@ void FFTSmooth2::exec() {
  */
 void FFTSmooth2::zero(int n, API::MatrixWorkspace_sptr &unfilteredWS,
                       API::MatrixWorkspace_sptr &filteredWS) {
-  int mx = static_cast<int>(unfilteredWS->readX(0).size());
-  int my = static_cast<int>(unfilteredWS->readY(0).size());
+  int mx = static_cast<int>(unfilteredWS->x(0).size());
+  int my = static_cast<int>(unfilteredWS->y(0).size());
   int ny = my / n;
 
   if (ny == 0)
     ny = 1;
 
   filteredWS =
-      API::WorkspaceFactory::Instance().create(unfilteredWS, 2, mx, my);
+	  API::WorkspaceFactory::Instance().create(unfilteredWS, 2, mx, my);
 
-  const Mantid::MantidVec &Yr = unfilteredWS->readY(0);
-  const Mantid::MantidVec &Yi = unfilteredWS->readY(1);
-  const Mantid::MantidVec &X = unfilteredWS->readX(0);
+  filteredWS->setSharedX(0, unfilteredWS->sharedX(0));
+  filteredWS->setSharedX(1, unfilteredWS->sharedX(0));
 
-  Mantid::MantidVec &yr = filteredWS->dataY(0);
-  Mantid::MantidVec &yi = filteredWS->dataY(1);
-  Mantid::MantidVec &xr = filteredWS->dataX(0);
-  Mantid::MantidVec &xi = filteredWS->dataX(1);
+  filteredWS->mutableY(0).assign(unfilteredWS->y(0).size(), 0);
+  filteredWS->mutableY(1).assign(unfilteredWS->y(0).size(), 0);
 
-  xr.assign(X.begin(), X.end());
-  xi.assign(X.begin(), X.end());
-  yr.assign(Yr.size(), 0);
-  yi.assign(Yr.size(), 0);
+  std::copy(unfilteredWS->y(0).cbegin(), unfilteredWS->y(0).begin() + ny,
+	  filteredWS->mutableY(0).begin());
+
+  std::copy(unfilteredWS->y(1).cbegin(), unfilteredWS->y(1).begin() + ny,
+	  filteredWS->mutableY(1).begin());
```

Comment:
@LamarMoore That is interesting, can you figure out why? Maybe there is a bug somewhere.